WordPress/Classes/ViewRelated/Blog/Blog Details/BlogDetailsViewController.m

```diff
@@ -524,7 +529,8 @@ - (void)showDetailViewForSubsection:(BlogDetailsSubsection)section
             [self.tableView selectRowAtIndexPath:indexPath
                                         animated:NO
                                   scrollPosition:[self optimumScrollPositionForIndexPath:indexPath]];
-            [self showMediaLibraryFromSource:BlogDetailsNavigationSourceLink];
+            BOOL showPicker = userInfo[[BlogDetailsViewController userInfoShowPickerKey]] != NULL;
```

Comment:
```suggestion
            BOOL showPicker = (userInfo[[BlogDetailsViewController userInfoShowPickerKey]] ?: NO) && self.blog.userCanUploadMedia;
```

- I feel `userInfo[[BlogDetailsViewController userInfoShowPickerKey]] ?: NO` is more semantically correct than `userInfo[[BlogDetailsViewController userInfoShowPickerKey]] != NULL`, wdyt?
- If a user can't upload media, let's show the media screen instead

Reply:
Addressed both comments. Nice catch with the upload eligbiility bug.